src/Concordium/Client/Types/Contract/Schema.hs

```diff
@@ -289,15 +385,19 @@ data FuncName
 
 -- |Try to find and decode an embedded `ModuleSchema` and a list of exported function
 -- names from inside a Wasm module.
-getEmbeddedSchemaAndExportsFromModule :: S.Get (Maybe ModuleSchema, [Text])
-getEmbeddedSchemaAndExportsFromModule = do
+getEmbeddedSchemaAndExportsFromModule :: Wasm.WasmVersion -> S.Get (Maybe ModuleSchema, [Text])
+getEmbeddedSchemaAndExportsFromModule wasmVersion = do
   mhBs <- S.getByteString 4
   unless (mhBs == wasmMagicHash) $ fail "Unknown magic value. This is likely not a Wasm module."
   vBs <- S.getByteString 4
-  unless (vBs == wasmVersion) $ fail "Unsupported Wasm version."
+  unless (vBs == wasmVersion0) $ fail "Unsupported Wasm version."
```

Comment:
I think this is confusingly named, since this wasm version is the version of the wasm standard, and is different from wasmVersion we use elsewhere. I think this should be renamed to `wasmStandardVersion` or some such.